src/networking.rs

```diff
@@ -138,22 +138,37 @@ pub struct NetworkingError {
     pub message: String,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ResponseRef {
+    response_ptr: usize,
+    response_len: usize,
+}
+impl ResponseRef {
+    pub fn new(response_ptr: usize, response_len: usize) -> Self {
```

Comment:
```suggestion
    pub fn new() -> Self {
```
这里不要参数吧，直接生成一个ref,反正ptr和len我们在这里没有用